torch/jit/_recursive.py

```diff
@@ -263,15 +268,17 @@ def infer_type(name, item):
             continue
 
         # If we got here, this is a regular "data" attribute, Add it to the concrete type
-        attr_type = infer_type(name, value)
-        if attr_type is not None:
-            concrete_type_builder.add_attribute(name, attr_type, False, False)
+        attr_type, inferred = infer_type(name, value)
+        if attr_type.success():
+            concrete_type_builder.add_attribute(name, attr_type.type(), False, False)
         else:
             # TODO: could add more detail here. For example, what the user should do
             # when the pytype is `list` or `NoneType`
-            hint = ("(This attribute exists on the Python module, "
-                    "but we failed to convert Python type: '{}' "
-                    "to a TorchScript type.)").format(torch.typename(type(value)))
+            inferred_msg = "Its type was inferred; try adding a type annotation for the attribute." if inferred else ""
```

Comment:
I commented about this on the issue but I wasn't able to come up with a test case that can help test this. Under what circumstances does an attribute fail to get converted without an annotation?